YoutubeDownloader/Views/Components/DashboardView.xaml.cs

```diff
@@ -1,23 +1,41 @@
-﻿using System.Windows.Input;
+using Avalonia.Controls;
+using Avalonia.Input;
+using Avalonia.Interactivity;
+using YoutubeDownloader.ViewModels.Components;
+using YoutubeDownloader.Views.Framework;
 
 namespace YoutubeDownloader.Views.Components;
 
-public partial class DashboardView
+public partial class DashboardView : ViewModelAwareUserControl<DashboardViewModel>
 {
     public DashboardView()
     {
         InitializeComponent();
+        QueryTextBox.AddHandler(
```

Comment:
Why is this event handled with `AddHandler` and the other (`OnStatusPointerReleased`) with XAML binding. Is one approach better than the other somehow? If they're the same, can we standardize?

Reply:
AddHanlder is used here as in Avalonia there are no Preview* helper events, so with using AddHandler you can specify the routing strategy as tunnel, which is the equivalent of preview events in WPF. A "preview" event is required here, as otherwise the enter key would have already created a new line.